app/assets/javascripts/pageflow/editor/models/file_uploader.js

```diff
@@ -15,13 +17,31 @@ pageflow.FileUploader = pageflow.Object.extend({
       fileType: fileType
     });
 
+    var uploadNestedFile = editor.nextUploadTargetFile;
+
+    if (uploadNestedFile){
+      if (fileType.topLevelType) {
+        throw(new pageflow.TopLevelTypeError(upload));
+      }
+      file.set({parent_file_id: editor.nextUploadTargetFile.get('id'),
+                parent_file_model_type: editor.nextUploadTargetFile.fileType().typeName});
+    }
+    else if (!fileType.topLevelType) {
+      throw(new pageflow.NestedTypeError(upload));
+    }
+
```

Comment:
This still does not check whether the the uploaded file is allowed as a nested file of the current targe.